src/core/dev_api/src/augru_cell.cpp

```diff
@@ -0,0 +1,149 @@
+// Copyright (C) 2018-2022 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "augru_cell.hpp"
+
+#include <cmath>
+
+#include "itt.hpp"
+#include "openvino/op/constant.hpp"
+#include "openvino/core/shape.hpp"
+#include "openvino/core/type/element_type.hpp"
+
+using namespace std;
+using namespace ov;
+
+BWDCMP_RTTI_DEFINITION(ov::op::v0::AUGRUCell);
+
+ov::op::v0::AUGRUCell::AUGRUCell() : m_linear_before_reset(false) {
+    m_activations = {"sigmoid", "tanh"};
+    m_activation_f = get_activation_function(0);
+    m_activation_g = get_activation_function(1);
+}
+
+ov::op::v0::AUGRUCell::AUGRUCell(const Output<Node>& X,
+                            const Output<Node>& H_t,
+                            const Output<Node>& W,
+                            const Output<Node>& R,
+                            const Output<Node>& B,
+                            const Output<Node>& A,
+                            size_t hidden_size)
+    : RNNCellBase({X, H_t, W, R, B, A}, hidden_size, 0.f, std::vector<std::string>{"sigmoid", "tanh"}, {}, {}),
+    m_activation_f{get_activation_function(0)},
+    m_activation_g{get_activation_function(1)},
+    m_linear_before_reset{false} {
+    constructor_validate_and_infer_types();
+}
+
+bool ov::op::v0::AUGRUCell::visit_attributes(AttributeVisitor& visitor) {
+    NGRAPH_OP_SCOPE(v0_AUGRUCell_visit_attributes);
+    visitor.on_attribute("linear_before_reset", m_linear_before_reset);
+    return op::util::RNNCellBase::visit_attributes(visitor);
+}
+
+void ov::op::v0::AUGRUCell::validate_and_infer_types() {
```

Comment:
@ceciliapeng2011 Because AUGRUCell/AUGRUSequence inherits from `RNNCellBase` as almost all sequence ops, moving to  `shape_inference` affects all of them. If we want to go this direction (@evgeniya-egupova please share your opinion), I am proposing to do it as a separate PR (here we are introducing a new cmake target, so it is not very clear to read)

Reply:
sounds good. okay, we could do it later.